raylutz/daffodil
# Fix package creation configuration

This change contains:
-  package configuration using `pyproject.toml` instead of `setup.py` because the standard and recommended method is `pyproject.toml`
- reorganize structure of code to have better build using the widely used project structuring
- renaming of imports to work for the new structure as well as in a project that imports daffodil
- removed a duplicated test in the main folder

There might be tests which are broken please check.

**NOTE** The versions I used contains rc1 to avoid name collision when publishing. 

### How To Test
1. First build the package locally using `python3 -m build`
2. Publish the package to PyPi using `python3 -m twine upload dist/*`. Make sure the `dist` folder only contains the assets of the latest build.
3. Install daffodil (in the project using daffodil) with the exact version just published. For example the existing version if built and uploaded directly will can be installed using `pip install daffodil==0.3.0rc1`
4. Run test commands and the project should run without `ModuleNotFoundError` for daffodil

Once verified that everything is working you can change the version in `pyproject.toml` to `0.3.1`, merge this to main branch then finally build and publish the package again.

For more info regarding how to create packages
- https://packaging.python.org/en/latest/tutorials/packaging-projects/
- https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html
- https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#writing-pyproject-toml
